sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -163,11 +163,13 @@ joint_matrix_fill(Group sg,
   std::ignore = sg;
   res.cuda_impl.wi_marray = v;
 #else
+  using storage_element_type =
+      typename oneapi::detail::helper_traits<T>::storage_element_type;
   res.spvm =
-      __spirv_CompositeConstruct<T, NumRows, NumCols,
+      __spirv_CompositeConstruct<storage_element_type, T, NumRows, NumCols,
                                  spv_matrix_use_traits<Use>::value,
                                  spv_matrix_layout_traits<Layout>::value>(
-          static_cast<T>(v));
+          static_cast<storage_element_type>(v));
```

Comment:
Are you sure this works? I just imagined that storage type could represent more than one element of size T2. But I do not know the details of the spirv case so just wanted to check this is fine.